lib/src/path_map.dart

```diff
@@ -7,32 +7,32 @@ import 'dart:collection';
 import '../path.dart' as p;
 
 /// A map whose keys are paths, compared using [equals] and [hash].
-class PathMap<V> extends MapView<String, V> {
+class PathMap<V> extends MapView<String?, V> {
   /// Creates an empty [PathMap] whose keys are compared using `context.equals`
   /// and `context.hash`.
   ///
   /// The [context] defaults to the current path context.
-  PathMap({p.Context context}) : super(_create(context));
+  PathMap({p.Context? context}) : super(_create(context));
 
   /// Creates a [PathMap] with the same keys and values as [other] whose keys
   /// are compared using `context.equals` and `context.hash`.
   ///
   /// The [context] defaults to the current path context. If multiple keys in
   /// [other] represent the same logical path, the last key's value will be
   /// used.
-  PathMap.of(Map<String, V> other, {p.Context context})
+  PathMap.of(Map<String, V> other, {p.Context? context})
       : super(_create(context)..addAll(other));
 
   /// Creates a map that uses [context] for equality and hashing.
-  static Map<String, V> _create<V>(p.Context context) {
+  static Map<String?, V> _create<V>(p.Context? context) {
```

Comment:
Needed testing to figure out I needed to add `?` to `Map<String, V>` here. Not sure how we could make it better.